proto/google/events/cloud/build/v1/data.proto

```diff
@@ -0,0 +1,626 @@
+  map<string, bytes> secret_env = 3;
+}
+
+// Optional arguments to enable specific features of builds.
+message BuildOptions {
+  // Specifies the manner in which the build should be verified, if at all.
+  enum VerifyOption {
+    // Not a verifiable build. (default)
+    NOT_VERIFIED = 0;
+
+    // Verified build.
+    VERIFIED = 1;
+  }
+
+  // Supported VM sizes.
+  enum MachineType {
+    // Standard machine type.
+    UNSPECIFIED = 0;
+
+    // Highcpu machine with 8 CPUs.
+    N1_HIGHCPU_8 = 1;
+
+    // Highcpu machine with 32 CPUs.
+    N1_HIGHCPU_32 = 2;
+  }
+
+  // Specifies the behavior when there is an error in the substitution checks.
+  enum SubstitutionOption {
+    // Fails the build if error in substitutions checks, like missing
+    // a substitution in the template or in the map.
+    MUST_MATCH = 0;
+
+    // Do not fail the build if error in substitutions checks.
+    ALLOW_LOOSE = 1;
+  }
+
+  // Specifies the behavior when writing build logs to Google Cloud Storage.
+  enum LogStreamingOption {
+    // Service may automatically determine build log streaming behavior.
+    STREAM_DEFAULT = 0;
+
+    // Build logs should be streamed to Google Cloud Storage.
+    STREAM_ON = 1;
+
+    // Build logs should not be streamed to Google Cloud Storage; they will be
+    // written when the build is completed.
+    STREAM_OFF = 2;
+  }
+
+  // Specifies the logging mode.
+  enum LoggingMode {
+    // The service determines the logging mode. The default is `LEGACY`. Do not
+    // rely on the default logging behavior as it may change in the future.
+    LOGGING_UNSPECIFIED = 0;
+
+    // Stackdriver logging and Cloud Storage logging are enabled.
+    LEGACY = 1;
+
+    // Only Cloud Storage logging is enabled.
+    GCS_ONLY = 2;
+  }
+
+  // Requested hash for SourceProvenance.
+  repeated Hash.HashType source_provenance_hash = 1;
+
+  // Requested verifiability options.
+  VerifyOption requested_verify_option = 2;
+
+  // Compute Engine machine type on which to run the build.
+  MachineType machine_type = 3;
+
+  // Requested disk size for the VM that runs the build. Note that this is *NOT*
+  // "disk free"; some of the space will be used by the operating system and
+  // build utilities. Also note that this is the minimum disk size that will be
+  // allocated for the build -- the build may run with a larger disk than
+  // requested. At present, the maximum disk size is 1000GB; builds that request
+  // more than the maximum are rejected with an error.
+  int64 disk_size_gb = 6;
+
+  // Option to specify behavior when there is an error in the substitution
+  // checks.
+  SubstitutionOption substitution_option = 4;
+
+  // Option to define build log streaming behavior to Google Cloud
+  // Storage.
+  LogStreamingOption log_streaming_option = 5;
+
+  // Option to specify a `WorkerPool` for the build.
+  // Format: projects/{project}/locations/{location}/workerPools/{workerPool}
+  //
+  // This field is experimental.
```

Comment:
Remove "experimental" part? (Not that it should be in the original API either, admittedly... stable APIs should be stable...)